tests/kafkatest/tests/core/group_mode_transactions_test.py

```diff
@@ -0,0 +1,312 @@
+                                  num_nodes=self.num_brokers,
+                                  zk=self.zk)
+
+    def setUp(self):
+        self.zk.start()
+
+    def seed_messages(self, topic, num_seed_messages):
+        seed_timeout_sec = 10000
+        seed_producer = VerifiableProducer(context=self.test_context,
+                                           num_nodes=1,
+                                           kafka=self.kafka,
+                                           topic=topic,
+                                           message_validator=is_int,
+                                           max_messages=num_seed_messages,
+                                           enable_idempotence=True,
+                                           repeating_keys=self.num_input_partitions)
+        seed_producer.start()
+        wait_until(lambda: seed_producer.num_acked >= num_seed_messages,
+                   timeout_sec=seed_timeout_sec,
+                   err_msg="Producer failed to produce messages %d in  %ds." % \
+                           (self.num_seed_messages, seed_timeout_sec))
+        return seed_producer.acked_by_partition
+
+    def get_messages_from_topic(self, topic, num_messages):
+        consumer = self.start_consumer(topic, group_id="verifying_consumer")
+        return self.drain_consumer(consumer, num_messages)
+
+    def bounce_brokers(self, clean_shutdown):
+        for node in self.kafka.nodes:
+            if clean_shutdown:
+                self.kafka.restart_node(node, clean_shutdown = True)
+            else:
+                self.kafka.stop_node(node, clean_shutdown = False)
+                wait_until(lambda: len(self.kafka.pids(node)) == 0 and not self.kafka.is_registered(node),
+                           timeout_sec=self.kafka.zk_session_timeout + 5,
+                           err_msg="Failed to see timely deregistration of \
+                           hard-killed broker %s" % str(node.account))
+                self.kafka.start_node(node)
+
+    def create_and_start_message_copier(self, input_topic, output_topic, transactional_id):
+        message_copier = TransactionalMessageCopier(
+            context=self.test_context,
+            num_nodes=1,
+            kafka=self.kafka,
+            transactional_id=transactional_id,
+            consumer_group=self.consumer_group,
+            input_topic=input_topic,
+            input_partition=-1,
+            output_topic=output_topic,
+            max_messages=-1,
+            transaction_size=self.transaction_size,
+            transaction_timeout=self.transaction_timeout,
+            group_mode=True
+        )
+        message_copier.start()
+        wait_until(lambda: message_copier.alive(message_copier.nodes[0]),
+                   timeout_sec=10,
+                   err_msg="Message copier failed to start after 10 s")
+        return message_copier
+
+    def bounce_copiers(self, copiers, clean_shutdown, timeout_sec=60):
```

Comment:
This timeout was increased from 30 to 60 seconds for the sake of rebalance caused offset wait (if triggered twice, that would be roughly 20~30 seconds)